lib/agent0/agent0/hyperdrive/exec/setup_experiment.py

```diff
@@ -1,41 +1,46 @@
 """Setup helper function for running eth bot experiments."""
 from __future__ import annotations
 
-import os
 from http import HTTPStatus
 
 import numpy as np
 import requests
-from agent0.base.config import EnvironmentConfig
+from agent0 import AccountKeyConfig
+from agent0.base.config import AgentConfig, EnvironmentConfig
 from agent0.hyperdrive.agents import HyperdriveAgent
-from agent0.hyperdrive.config import get_eth_bots_config
 from agent0.hyperdrive.crash_report import setup_hyperdrive_crash_report_logging
-from agent0.hyperdrive.exec import get_agent_accounts
 from elfpy.utils import logs
+from ethpy import EthConfig
 from ethpy.base import initialize_web3_with_http_provider, load_all_abis
-from ethpy.hyperdrive import fetch_hyperdrive_address_from_url
+from ethpy.hyperdrive.addresses import HyperdriveAddresses
 from web3 import Web3
 from web3.contract.contract import Contract
 
+from .get_agent_accounts import get_agent_accounts
 
-def setup_experiment() -> tuple[Web3, Contract, Contract, EnvironmentConfig, list[HyperdriveAgent]]:
+
+def setup_experiment(
+    eth_config: EthConfig,
+    environment_config: EnvironmentConfig,
+    agent_config: list[AgentConfig],
+    account_key_config: AccountKeyConfig,
+    contract_addresses: HyperdriveAddresses,
+) -> tuple[Web3, Contract, list[HyperdriveAgent]]:
     """Get agents according to provided config, provide eth, base token and approve hyperdrive.
 
     Returns
     -------
     tuple[Web3, Contract, Contract, EnvironmentConfig, list[HyperdriveAgent]]
         A tuple containing:
             - The web3 container
-            - The base token contract
```

Comment:
We were returning both contracts because we kept needing them for debugging. Maybe the solution here (instead of returning an unused variable) is to return a dict with the key corresponding to the contract name and the value the contract itself. You can use the same key as was used to access the ABIs:`abis[environment_config.base_abi]`